src/screens/AppIntroScreen.tsx

```diff
@@ -158,69 +246,60 @@ export const AppIntroScreen = ({ setOnboardingComplete, onlyTermsAndConditions }
+        onSkip={handleSkip}
+        ref={slider}
+        renderNextButton={() => {
+          const currentSlideIndex = slider.current?.state?.activeIndex ?? 0;
+          const isTermsSlide =
+            slides[currentSlideIndex]?.onLeaveSlideName === Initializer.TermsAndConditions;
+
+          const buttonStyle =
+            isTermsSlide && !hasAcceptedTerms
+              ? [styles.sliderButtonContainer, styles.sliderButtonDisabled]
+              : styles.sliderButtonContainer;
+
+          return (
+            <TouchableOpacity
+              onPress={() => {
+                if (isTermsSlide && !hasAcceptedTerms) {
+                  termsAndConditionsAlert();
+
+                  return;
+                }
+
+                slider.current?.goToSlide(currentSlideIndex + 1, true);
+              }}
+            >
+              <View style={buttonStyle}>
+                <BoldText lightest={!(isTermsSlide && !hasAcceptedTerms)}>
+                  {texts.appIntro.continue.toUpperCase()}
+                </BoldText>
+              </View>
+            </TouchableOpacity>
+          );
         }}
-        renderDoneButton={() => <SliderButton label={texts.appIntro.continue} />}
-        renderItem={({ item }) =>
-          renderSlide({
-            item,
-            setShowButtonTermsAndConditions: (value) => {
-              setShowNextButtonTermsAndConditions(value);
-              setShowDoneButtonTermsAndConditions(value);
-            }
-          })
-        }
-        renderNextButton={() => <SliderButton label={texts.appIntro.continue} />}
+        renderSkipButton={() => (
+          <SliderButton label={texts.appIntro.skip} style={styles.skipButton} />
+        )}
         scrollEnabled={false}
         showDoneButton={showDoneButtonTermsAndConditions}
-        showNextButton={showNextButtonTermsAndConditions}
+        showNextButton
+        showSkipButton
         style={device.platform === 'android' && { paddingTop: getStatusBarHeight() }}
       />
-      {(!showNextButtonTermsAndConditions || !showDoneButtonTermsAndConditions) && (
-        <View
-          style={[styles.termsAndConditionsNextButtonContainer, { bottom: safeAreaBottom + 16 }]}
-        >
-          <View style={styles.termsAndConditionsNextButton}>
-            <TouchableOpacity
-              onPress={() =>
-                Alert.alert(
-                  texts.profile.termsAndConditionsAlertTitle,
-                  texts.profile.termsAndConditionsAlertMessage,
-                  [
-                    {
-                      text: texts.profile.termsAndConditionsAlertOk
-                    }
-                  ]
-                )
-              }
-            >
-              <SliderButton
-                isPlaceholder
-                label={texts.appIntro.continue}
-                style={styles.sliderButtonAdditional}
-              />
-            </TouchableOpacity>
-          </View>
-        </View>
-      )}
```

Comment:
how come all of this could be removed?

Reply:
Because now we create the buttons through the props in `AppIntreSlider`.